### ISSUE TITLE
"Lock file" error on "Step 10/19" on `make patch images`

### ISSUE THREAD

**Issue by mkasprz** [alswl/excalidraw-collaboration#18]
Hi.

After running `make patch images`, on "Step 10/19", an error appears:

```
Step 10/19 : RUN npm ci --prod
 ---> Running in 950bdcb000bc
npm WARN config production Use `--omit=dev` instead.
npm ERR! code EUSAGE
npm ERR! 
npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm ERR! 
npm ERR! Invalid: lock file's sqlite3@5.0.2 does not satisfy sqlite3@5.0.3
npm ERR! Missing: @mapbox/node-pre-gyp@1.0.11 from lock file
npm ERR! Invalid: lock file's node-addon-api@3.2.1 does not satisfy node-addon-api@4.3.0
npm ERR! Invalid: lock file's node-gyp@3.8.0 does not satisfy node-gyp@8.4.1
npm ERR! Invalid: lock file's tar@2.2.2 does not satisfy tar@6.1.15
npm ERR! Invalid: lock file's detect-libc@1.0.3 does not satisfy detect-libc@2.0.2
npm ERR! Invalid: lock file's node-fetch@2.6.6 does not satisfy node-fetch@2.6.12
npm ERR! Invalid: lock file's nopt@3.0.6 does not satisfy nopt@5.0.0
npm ERR! Invalid: lock file's npmlog@4.1.2 does not satisfy npmlog@5.0.1
npm ERR! Missing: env-paths@2.2.1 from lock file
npm ERR! Missing: make-fetch-happen@9.1.0 from lock file
npm ERR! Missing: npmlog@6.0.2 from lock file
npm ERR! Missing: agentkeepalive@4.3.0 from lock file
npm ERR! Missing: cacache@15.3.0 from lock file
npm ERR! Missing: http-cache-semantics@4.1.1 from lock file
npm ERR! Missing: is-lambda@1.0.1 from lock file
npm ERR! Invalid: lock file's minipass@2.9.0 does not satisfy minipass@3.3.6
npm ERR! Missing: minipass-collect@1.0.2 from lock file
npm ERR! Missing: minipass-fetch@1.4.1 from lock file
npm ERR! Missing: minipass-flush@1.0.5 from lock file
npm ERR! Missing: minipass-pipeline@1.2.4 from lock file
npm ERR! Missing: promise-retry@2.0.1 from lock file
npm ERR! Missing: socks-proxy-agent@6.2.1 from lock file
npm ERR! Missing: ssri@8.0.1 from lock file
npm ERR! Missing: depd@2.0.0 from lock file
npm ERR! Missing: humanize-ms@1.2.1 from lock file
npm ERR! Missing: @npmcli/fs@1.1.1 from lock file
npm ERR! Missing: @npmcli/move-file@1.1.2 from lock file
npm ERR! Invalid: lock file's chownr@1.1.4 does not satisfy chownr@2.0.0
npm ERR! Invalid: lock file's fs-minipass@1.2.7 does not satisfy fs-minipass@2.1.0
npm ERR! Missing: infer-owner@1.0.4 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: p-map@4.0.0 from lock file
npm ERR! Missing: promise-inflight@1.0.1 from lock file
npm ERR! Missing: unique-filename@1.1.1 from lock file
npm ERR! Missing: @gar/promisify@1.1.3 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: encoding@0.1.13 from lock file
npm ERR! Missing: minipass-sized@1.0.3 from lock file
npm ERR! Invalid: lock file's minizlib@1.3.3 does not satisfy minizlib@2.1.2
npm ERR! Missing: iconv-lite@0.6.3 from lock file
npm ERR! Invalid: lock file's are-we-there-yet@1.1.7 does not satisfy are-we-there-yet@2.0.0
npm ERR! Invalid: lock file's gauge@2.7.4 does not satisfy gauge@3.0.2
npm ERR! Invalid: lock file's aproba@1.2.0 does not satisfy aproba@2.0.0
npm ERR! Missing: color-support@1.1.3 from lock file
npm ERR! Missing: err-code@2.0.3 from lock file
npm ERR! Missing: retry@0.12.0 from lock file
npm ERR! Invalid: lock file's debug@4.3.2 does not satisfy debug@4.3.4
npm ERR! Missing: socks@2.7.1 from lock file
npm ERR! Missing: ip@2.0.0 from lock file
npm ERR! Missing: smart-buffer@4.2.0 from lock file
npm ERR! Missing: minipass@5.0.0 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: unique-slug@2.0.2 from lock file
npm ERR! Missing: aggregate-error@3.1.0 from lock file
npm ERR! Missing: clean-stack@2.2.0 from lock file
npm ERR! Missing: indent-string@4.0.0 from lock file
npm ERR! Missing: are-we-there-yet@3.0.1 from lock file
npm ERR! Missing: gauge@4.0.4 from lock file
npm ERR! Invalid: lock file's signal-exit@3.0.6 does not satisfy signal-exit@3.0.7
npm ERR! 
npm ERR! Clean install a project
npm ERR! 
npm ERR! Usage:
npm ERR! npm ci
npm ERR! 
npm ERR! Options:
npm ERR! [-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]
npm ERR! [-E|--save-exact] [-g|--global] [--global-style] [--legacy-bundling]
npm ERR! [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
npm ERR! [--strict-peer-deps] [--no-package-lock] [--foreground-scripts]
npm ERR! [--ignore-scripts] [--no-audit] [--no-bin-links] [--no-fund] [--dry-run]
npm ERR! [-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
npm ERR! [-ws|--workspaces] [--include-workspace-root] [--install-links]
npm ERR! 
npm ERR! aliases: clean-install, ic, install-clean, isntall-clean
npm ERR! 
npm ERR! Run "npm help ci" for more info

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-07-27T21_27_03_382Z-debug-0.log
The command '/bin/sh -c npm ci --prod' returned a non-zero code: 1
```

<details><summary><b>More of the output</b></summary>
<p>

```
# you can edit excalidraw.env.production to change endpoints
cp excalidraw.env.production excalidraw/.env.production
cp excalidraw.dockerignore excalidraw/.dockerignore
Sending build context to Docker daemon    6.9MB
Step 1/10 : FROM node:14-alpine AS build
 ---> 0dac3dc27b1a
Step 2/10 : WORKDIR /opt/node_app
 ---> Using cache
 ---> 25b5736578cd
Step 3/10 : COPY package.json yarn.lock ./
 ---> Using cache
 ---> 4b552c2173c8
Step 4/10 : RUN yarn --ignore-optional --network-timeout 600000
 ---> Using cache
 ---> 229c9e84bb97
Step 5/10 : ARG NODE_ENV=production
 ---> Using cache
 ---> 6888937c6459
Step 6/10 : COPY . .
 ---> Using cache
 ---> 2b48cefe3a5e
Step 7/10 : RUN yarn build:app:docker
 ---> Using cache
 ---> 9e6b5d2a64e4
Step 8/10 : FROM nginx:1.21-alpine
 ---> b1c3acb28882
Step 9/10 : COPY --from=build /opt/node_app/build /usr/share/nginx/html
 ---> Using cache
 ---> 7fbdc91daeb1
Step 10/10 : HEALTHCHECK CMD wget -q -O /dev/null http://localhost || exit 1
 ---> Using cache
 ---> e2322dc297d6
[Warning] One or more build-args [COMMIT] were not consumed
Successfully built e2322dc297d6
Successfully tagged excalidraw:v0.1.0-6c3da92
Sending build context to Docker daemon    298kB
Step 1/9 : FROM node:12-alpine
 ---> bb6d28039b8c
Step 2/9 : WORKDIR /excalidraw-room
 ---> Using cache
 ---> 32e137a17b7c
Step 3/9 : COPY package.json yarn.lock ./
 ---> Using cache
 ---> 11aad932c141
Step 4/9 : RUN yarn
 ---> Using cache
 ---> 27c3841c4608
Step 5/9 : COPY tsconfig.json ./
 ---> Using cache
 ---> 9497a88b95bb
Step 6/9 : COPY src ./src
 ---> Using cache
 ---> dc51d8e9565d
Step 7/9 : RUN yarn build
 ---> Using cache
 ---> 460c002d1b5a
Step 8/9 : EXPOSE 80
 ---> Using cache
 ---> 426ec4a215fe
Step 9/9 : CMD ["yarn", "start"]
 ---> Using cache
 ---> 4aa3c3d44036
[Warning] One or more build-args [COMMIT] were not consumed
Successfully built 4aa3c3d44036
Successfully tagged excalidraw-room:v0.1.0-6c3da92
Sending build context to Docker daemon  822.3kB
Step 1/19 : FROM node:16-alpine as builder
 ---> d22e42cbfdbb
Step 2/19 : RUN apk add --update python3 make g++
 ---> Using cache
 ---> 5f1f7db154b3
Step 3/19 : RUN npm install -g eslint
 ---> Using cache
 ---> 6ca5fe7af870
Step 4/19 : RUN npm install -g @nestjs/cli
 ---> Using cache
 ---> fb9ec2c7bb45
Step 5/19 : WORKDIR /app
 ---> Using cache
 ---> 5258004fa2a5
Step 6/19 : COPY package.json .
 ---> Using cache
 ---> d987d0c6c618
Step 7/19 : COPY package-lock.json .
 ---> Using cache
 ---> b562279862ee
Step 8/19 : RUN npm install
 ---> Using cache
 ---> 0d81d5692ac7
Step 9/19 : COPY . .
 ---> Using cache
 ---> 5a7e4ead41b3
Step 10/19 : RUN npm ci --prod
 ---> Running in 950bdcb000bc
npm WARN config production Use `--omit=dev` instead.
npm ERR! code EUSAGE
npm ERR! 
npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm ERR! 
npm ERR! Invalid: lock file's sqlite3@5.0.2 does not satisfy sqlite3@5.0.3
npm ERR! Missing: @mapbox/node-pre-gyp@1.0.11 from lock file
npm ERR! Invalid: lock file's node-addon-api@3.2.1 does not satisfy node-addon-api@4.3.0
npm ERR! Invalid: lock file's node-gyp@3.8.0 does not satisfy node-gyp@8.4.1
npm ERR! Invalid: lock file's tar@2.2.2 does not satisfy tar@6.1.15
npm ERR! Invalid: lock file's detect-libc@1.0.3 does not satisfy detect-libc@2.0.2
npm ERR! Invalid: lock file's node-fetch@2.6.6 does not satisfy node-fetch@2.6.12
npm ERR! Invalid: lock file's nopt@3.0.6 does not satisfy nopt@5.0.0
npm ERR! Invalid: lock file's npmlog@4.1.2 does not satisfy npmlog@5.0.1
npm ERR! Missing: env-paths@2.2.1 from lock file
npm ERR! Missing: make-fetch-happen@9.1.0 from lock file
npm ERR! Missing: npmlog@6.0.2 from lock file
npm ERR! Missing: agentkeepalive@4.3.0 from lock file
npm ERR! Missing: cacache@15.3.0 from lock file
npm ERR! Missing: http-cache-semantics@4.1.1 from lock file
npm ERR! Missing: is-lambda@1.0.1 from lock file
npm ERR! Invalid: lock file's minipass@2.9.0 does not satisfy minipass@3.3.6
npm ERR! Missing: minipass-collect@1.0.2 from lock file
npm ERR! Missing: minipass-fetch@1.4.1 from lock file
npm ERR! Missing: minipass-flush@1.0.5 from lock file
npm ERR! Missing: minipass-pipeline@1.2.4 from lock file
npm ERR! Missing: promise-retry@2.0.1 from lock file
npm ERR! Missing: socks-proxy-agent@6.2.1 from lock file
npm ERR! Missing: ssri@8.0.1 from lock file
npm ERR! Missing: depd@2.0.0 from lock file
npm ERR! Missing: humanize-ms@1.2.1 from lock file
npm ERR! Missing: @npmcli/fs@1.1.1 from lock file
npm ERR! Missing: @npmcli/move-file@1.1.2 from lock file
npm ERR! Invalid: lock file's chownr@1.1.4 does not satisfy chownr@2.0.0
npm ERR! Invalid: lock file's fs-minipass@1.2.7 does not satisfy fs-minipass@2.1.0
npm ERR! Missing: infer-owner@1.0.4 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: p-map@4.0.0 from lock file
npm ERR! Missing: promise-inflight@1.0.1 from lock file
npm ERR! Missing: unique-filename@1.1.1 from lock file
npm ERR! Missing: @gar/promisify@1.1.3 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: encoding@0.1.13 from lock file
npm ERR! Missing: minipass-sized@1.0.3 from lock file
npm ERR! Invalid: lock file's minizlib@1.3.3 does not satisfy minizlib@2.1.2
npm ERR! Missing: iconv-lite@0.6.3 from lock file
npm ERR! Invalid: lock file's are-we-there-yet@1.1.7 does not satisfy are-we-there-yet@2.0.0
npm ERR! Invalid: lock file's gauge@2.7.4 does not satisfy gauge@3.0.2
npm ERR! Invalid: lock file's aproba@1.2.0 does not satisfy aproba@2.0.0
npm ERR! Missing: color-support@1.1.3 from lock file
npm ERR! Missing: err-code@2.0.3 from lock file
npm ERR! Missing: retry@0.12.0 from lock file
npm ERR! Invalid: lock file's debug@4.3.2 does not satisfy debug@4.3.4
npm ERR! Missing: socks@2.7.1 from lock file
npm ERR! Missing: ip@2.0.0 from lock file
npm ERR! Missing: smart-buffer@4.2.0 from lock file
npm ERR! Missing: minipass@5.0.0 from lock file
npm ERR! Missing: mkdirp@1.0.4 from lock file
npm ERR! Missing: unique-slug@2.0.2 from lock file
npm ERR! Missing: aggregate-error@3.1.0 from lock file
npm ERR! Missing: clean-stack@2.2.0 from lock file
npm ERR! Missing: indent-string@4.0.0 from lock file
npm ERR! Missing: are-we-there-yet@3.0.1 from lock file
npm ERR! Missing: gauge@4.0.4 from lock file
npm ERR! Invalid: lock file's signal-exit@3.0.6 does not satisfy signal-exit@3.0.7
npm ERR! 
npm ERR! Clean install a project
npm ERR! 
npm ERR! Usage:
npm ERR! npm ci
npm ERR! 
npm ERR! Options:
npm ERR! [-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]
npm ERR! [-E|--save-exact] [-g|--global] [--global-style] [--legacy-bundling]
npm ERR! [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
npm ERR! [--strict-peer-deps] [--no-package-lock] [--foreground-scripts]
npm ERR! [--ignore-scripts] [--no-audit] [--no-bin-links] [--no-fund] [--dry-run]
npm ERR! [-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
npm ERR! [-ws|--workspaces] [--include-workspace-root] [--install-links]
npm ERR! 
npm ERR! aliases: clean-install, ic, install-clean, isntall-clean
npm ERR! 
npm ERR! Run "npm help ci" for more info

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-07-27T21_27_03_382Z-debug-0.log
The command '/bin/sh -c npm ci --prod' returned a non-zero code: 1
```
</p>
</details> 

It's Ubuntu 22.04 and Docker 20.10.21.

What could I try doing to resolve the issue?

**Comment by alswl** [comment]
Please try these commands, and find out which one cause problem.

```
make images TARGETS=excalidraw
make images TARGETS=excalidraw-room
make images TARGETS=excalidraw-storage-backend
```

**Comment by alswl** [comment]
Fixed.

**Comment by mkasprz** [comment]
> Please try these commands, and find out which one cause problem.
> 
> ```
> make images TARGETS=excalidraw
> make images TARGETS=excalidraw-room
> make images TARGETS=excalidraw-storage-backend
> ```
> 
> update: I found `make images TARGETS=excalidraw-storage-backend` produce the error, I'm working on it.

Hi. Sorry I didn't answer before.

> Fixed by #19 .

I confirm that the issue seems resolved. Thank You very much for Your help and all! I managed to have the instance up and running. :tada:
